### PR TITLE
Use decimal constructor

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo-modules/issues/472
Require https://github.com/neo-project/neo-modules/pull/477

### CODE REVIEW

**Review by devhawk — APPROVED** [review]

**Comment by chenzhitong** [comment]
Do these areas also need to be modified?

https://github.com/neo-project/neo-modules/blob/515608a54292d059e3c115c15fabf7c43ae00102/src/RpcClient/Models/RpcApplicationLog.cs#L71

https://github.com/neo-project/neo-modules/blob/515608a54292d059e3c115c15fabf7c43ae00102/src/RpcClient/TransactionManagerFactory.cs#L52

https://github.com/neo-project/neo-modules/blob/515608a54292d059e3c115c15fabf7c43ae00102/src/RpcClient/Utility.cs#L157-L158

File: src/RpcServer/RpcServerSettings.cs [hunk 1]
```diff
@@ -48,12 +47,19 @@ public class RpcServerSettings
...
+            this.MaxFee = ToBigDecimalGas(maxFee);
             this.DisabledMethods = disabledMethods ?? Array.Empty<string>();
             this.MaxConcurrentConnections = maxConcurrentConnections;
         }
 
+        private static long ToBigDecimalGas(decimal maxGasInvoke)
+        {
+            var dec = new BigDecimal(maxGasInvoke);
+            dec.ChangeDecimals(NativeContract.GAS.Decimals);
+            return (long)dec.Value;
```
Comment (erikzhang) [inline src/RpcServer/RpcServerSettings.cs:60]: We should add a new constructor to `BigDecimal`.

```c#
public BigDecimal(decimal value, byte decimals)
```

**Comment by shargon** [comment]
@chenzhitong Some of them comes from us, with a string with '.'

**Comment by shargon** [comment]
Wait for https://github.com/neo-project/neo-modules/pull/477

**Comment by devhawk** [comment]
Thanks for looking at other places we use BigDecimal.Parse @chenzhitong. We should not use ToString -> Parse. For JSON parsing, I think we are stuck using Parse.

**Review by erikzhang — APPROVED** [review]

**Pull request merged**